apps/faucet/src/App/Faucet.tsx

```diff
@@ -52,16 +38,41 @@ type Props = {
 
 export const FaucetForm: React.FC<Props> = ({ isTestnetLive }) => {
   const theme = useTheme();
+  const { difficulty, settingsError, limit, tokens, url } =
+    useContext(AppContext);
   const [targetAddress, setTargetAddress] = useState<string>();
-  const [tokenAddress, setTokenAddress] = useState(TokenData[0].value);
+  const [tokenAddress, setTokenAddress] = useState<string>();
   const [amount, setAmount] = useState<number | undefined>(undefined);
   const [error, setError] = useState<string>();
   const [status, setStatus] = useState(Status.Completed);
   const [statusText, setStatusText] = useState<string>();
   const [responseDetails, setResponseDetails] = useState<TransferResponse>();
 
+  useEffect(() => {
+    if (tokens?.NAM) {
+      setTokenAddress(tokens.NAM);
+    }
+  }, [tokens]);
+
+  const isFormValid = (): boolean => {
+    return (
+      Boolean(tokenAddress) &&
+      Boolean(amount) &&
+      (amount || 0) <= limit &&
+      Boolean(targetAddress) &&
+      status !== Status.Pending &&
+      isTestnetLive
+    );
+  };
+
   const handleSubmit = useCallback(async () => {
-    if (!targetAddress || !amount) {
+    if (
```

Comment:
That's true - this function should never be able to be called if invalid, and I added that solely because they can be `undefined` - I actually did try adding `isFormValid()` here, but still needed the check for values that are loaded async. I can take another look at it!